crates/topos/src/components/tce/services/push_certificate.rs

```diff
@@ -31,143 +31,27 @@ pub(crate) async fn check_delivery(
-                        "Unable to execute the watch_certificates on TCE api",
-                    )
-                })?;
-                let mut resp_stream = response.into_inner();
-                async move {
-                    while let Some(received) = resp_stream.next().await {
-                        let received = received.unwrap();
-                        if let Some(Event::CertificatePushed(CertificatePushed {
-                            certificate: Some(certificate),
-                            ..
-                        })) = received.event
-                        {
-                            // unwrap is safe because we are sure that the certificate is present
-                            if certificate_id == certificate.id.unwrap() {
-                                debug!("Received the certificate on {}", peer_string);
-                                return Ok(());
-                            }
-                        }
-                    }
-
-                    Err((peer_string.clone(), "didn't receive any certificate"))
-                }
-                .await
-            }));
-        }
-
-        let mut client = ApiServiceClient::connect(random_peer.clone())
-            .await
-            .map_err(|_| vec![format!("Unable to connect to the TCE api on {random_peer}")])?;
-
-        // submit a certificate to one node
-        _ = client
-            .submit_certificate(SubmitCertificateRequest {
-                certificate: Some(pushed_certificate.into()),
-            })
-            .await
-            .map_err(|_| {
-                vec![format!(
-                    "Unable to submit the certificate to the TCE api on {random_peer}"
-                )]
-            })?;
-
-        tokio::time::sleep(Duration::from_secs(timeout_broadcast)).await;
-        let mut errors = vec![];
-
-        join_all(join_handlers)
-            .await
-            .iter()
-            .for_each(|result| match result {
-                Err(_) => {
-                    errors.push("Unable to properly execute command".to_string());
-                }
-                Ok(Err((peer, error))) => {
-                    errors.push(format!("{peer} {error}"));
-                }
-                _ => {}
-            });
-
-        if errors.is_empty() {
-            Ok(())
-        } else {
-            Err(errors)
-        }
-    })
-    .await
-    .map_err(|error| {
-        info!("Timeout reached: {:?}", error);
-        error
-    })
+) -> Result<(), Box<dyn std::error::Error>> {
+    info!("peers: {:?}", peers);
+
+    let peers: Vec<Uri> = format
+        .parse(NodeList(peers))
+        .map_err(|e| format!("Unable to parse node list: {e}"))?
+        .into_iter()
+        .map(TryInto::try_into)
+        .collect::<Result<Vec<_>, _>>()
+        .map_err(|e| format!("Unable to parse node list: {e}"))?;
+
+    match topos_test_sdk::integration::check_certificate_delivery(timeout_broadcast, peers, timeout)
```

Comment:
`topos-test-sdk` as mentioned by its name, must only be use for tests, not for production code.

Reply:
This is not production code but test code

Reply:
It is part of `topos/src` then it's part of the binary used in production.
It means that we are pulling every dev-dependency (rstest, custom server implementation, mock) into the final binary).